### PR TITLE
Fix compile warnings

### PR DESCRIPTION
**Pull request by Joseph-Edwards**
This PR continues the work form #633 by fixing compiler warnings for GCC.

### CODE REVIEW

**Comment by james-d-mitchell** [comment]
Seems like the file is being reformatted which makes it a bit hard to see what the actual changes are. Also do we expect the ci to fail?

**Comment by Joseph-Edwards** [comment]
I've undone that formatting, so it should now be clearer what the changes are. I didn't expect the CI to fail, but since we are planning on dropping the support for GAP 4.10, can this be ignored or does it still warrant investigation? @james-d-mitchell

**Comment by Joseph-Edwards** [comment]
> I didn't expect the CI to fail, but since we are planning on dropping the support for GAP 4.10, can this be ignored or does it still warrant investigation?

In case there was a fundamental error, I decided to investigate. In `ci/docker-test.sh`, for version 4.10, GAPDoc is being cloned using
```
URL="http://www.math.rwth-aachen.de/~Frank.Luebeck/GAPDoc/GAPDoc-$VERSION.tar.gz"
```
but this link appears to be dead. This is what was causing the 4.10 tests to fail.

Changing the above to 
```
URL="https://github.com/frankluebeck/GAPDoc/archive/refs/tags/relv$VERSION.tar.gz"
```
seems to fix the problem. Since the 4.10 jobs will soon be dropped, it doesn't seem like there is any need to make the above change.

File: extern/edge-addition-planarity-suite-Version_3.0.1.0/c/graphDFSUtils.c [hunk 1]
```diff
@@ -31,6 +31,11 @@ extern void _ClearVertexVisitedFlags(graphP theGraph, int);
 #pragma clang diagnostic ignored "-Wstrict-prototypes"
 #pragma clang diagnostic ignored "-Wmissing-prototypes"
 #pragma clang diagnostic ignored "-Wdeprecated-non-prototype"
+#elif defined(__GNUC__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstrict-prototypes"
+#pragma GCC diagnostic ignored "-Wmissing-prototypes"
+
```
Comment (james-d-mitchell) [inline extern/edge-addition-planarity-suite-Version_3.0.1.0/c/graphDFSUtils.c:38]:
```suggestion
```

File: extern/edge-addition-planarity-suite-Version_3.0.1.0/c/graphDrawPlanar.c [hunk 2]
```diff
@@ -9,6 +9,11 @@ See the LICENSE.TXT file for licensing information.
 #pragma clang diagnostic ignored "-Wstrict-prototypes"
 #pragma clang diagnostic ignored "-Wmissing-prototypes"
 #pragma clang diagnostic ignored "-Wdeprecated-non-prototype"
+#elif defined(__GNUC__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstrict-prototypes"
+#pragma GCC diagnostic ignored "-Wmissing-prototypes"
+
```
Comment (james-d-mitchell) [inline extern/edge-addition-planarity-suite-Version_3.0.1.0/c/graphDrawPlanar.c:16]:
```suggestion
```

File: extern/edge-addition-planarity-suite-Version_3.0.1.0/c/graphDrawPlanar_Extensions.c [hunk 3]
```diff
@@ -3,20 +3,24 @@ Copyright (c) 1997-2020, John M. Boyer
...
+#pragma GCC diagnostic ignored "-Wmissing-prototypes"
+#endif
 
 #include <stdlib.h>
 
 #include "graphDrawPlanar.private.h"
 #include "graphDrawPlanar.h"
-
-
+
```
Comment (james-d-mitchell) [inline extern/edge-addition-planarity-suite-Version_3.0.1.0/c/graphDrawPlanar_Extensions.c:22]: Still formatting changes here

File: extern/edge-addition-planarity-suite-Version_3.0.1.0/c/graphDrawPlanar_Extensions.c [hunk 4]
```diff
@@ -59,7 +63,7 @@ void _DrawPlanar_FreeContext(void *);
 /****************************************************************************
  * DRAWPLANAR_ID - the variable used to hold the integer identifier for this
  * extension, enabling this feature's extension context to be distinguished
- * from other features' extension contexts that may be attached to a graph.
+ * from other features' extension contexts that may be attached to a graph.
```
Comment (james-d-mitchell) [inline extern/edge-addition-planarity-suite-Version_3.0.1.0/c/graphDrawPlanar_Extensions.c:66]: and here

File: extern/edge-addition-planarity-suite-Version_3.0.1.0/c/graphEmbed.c [hunk 5]
```diff
@@ -13,6 +13,11 @@ See the LICENSE.TXT file for licensing information.
 #pragma clang diagnostic ignored "-Wstrict-prototypes"
 #pragma clang diagnostic ignored "-Wmissing-prototypes"
 #pragma clang diagnostic ignored "-Wdeprecated-non-prototype"
+#elif defined(__GNUC__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstrict-prototypes"
+#pragma GCC diagnostic ignored "-Wmissing-prototypes"
+
```
Comment (james-d-mitchell) [inline extern/edge-addition-planarity-suite-Version_3.0.1.0/c/graphEmbed.c:20]:
```suggestion
```

File: extern/edge-addition-planarity-suite-Version_3.0.1.0/c/graphFunctionTable.h [hunk 6]
```diff
@@ -21,6 +21,11 @@ extern "C" {
 #pragma clang diagnostic ignored "-Wstrict-prototypes"
 #pragma clang diagnostic ignored "-Wmissing-prototypes"
 #pragma clang diagnostic ignored "-Wdeprecated-non-prototype"
+#elif defined(__GNUC__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstrict-prototypes"
+#pragma GCC diagnostic ignored "-Wmissing-prototypes"
+
```
Comment (james-d-mitchell) [inline extern/edge-addition-planarity-suite-Version_3.0.1.0/c/graphFunctionTable.h:28]:
```suggestion
```

File: extern/edge-addition-planarity-suite-Version_3.0.1.0/c/graphIsolator.c [hunk 7]
```diff
@@ -9,6 +9,11 @@ See the LICENSE.TXT file for licensing information.
 #pragma clang diagnostic ignored "-Wstrict-prototypes"
 #pragma clang diagnostic ignored "-Wmissing-prototypes"
 #pragma clang diagnostic ignored "-Wdeprecated-non-prototype"
+#elif defined(__GNUC__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstrict-prototypes"
+#pragma GCC diagnostic ignored "-Wmissing-prototypes"
+
```
Comment (james-d-mitchell) [inline extern/edge-addition-planarity-suite-Version_3.0.1.0/c/graphIsolator.c:16]:
```suggestion
```

File: extern/edge-addition-planarity-suite-Version_3.0.1.0/c/graphK23Search.c [hunk 8]
```diff
@@ -9,6 +9,11 @@ See the LICENSE.TXT file for licensing information.
 #pragma clang diagnostic ignored "-Wstrict-prototypes"
 #pragma clang diagnostic ignored "-Wmissing-prototypes"
 #pragma clang diagnostic ignored "-Wdeprecated-non-prototype"
+#elif defined(__GNUC__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstrict-prototypes"
+#pragma GCC diagnostic ignored "-Wmissing-prototypes"
+
```
Comment (james-d-mitchell) [inline extern/edge-addition-planarity-suite-Version_3.0.1.0/c/graphK23Search.c:16]:
```suggestion
```

File: extern/edge-addition-planarity-suite-Version_3.0.1.0/c/graphK33Search_Extensions.c [hunk 9]
```diff
@@ -9,6 +9,11 @@ See the LICENSE.TXT file for licensing information.
 #pragma clang diagnostic ignored "-Wstrict-prototypes"
 #pragma clang diagnostic ignored "-Wmissing-prototypes"
 #pragma clang diagnostic ignored "-Wdeprecated-non-prototype"
+#elif defined(__GNUC__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstrict-prototypes"
+#pragma GCC diagnostic ignored "-Wmissing-prototypes"
+
```
Comment (james-d-mitchell) [inline extern/edge-addition-planarity-suite-Version_3.0.1.0/c/graphK33Search_Extensions.c:16]:
```suggestion
```

File: extern/edge-addition-planarity-suite-Version_3.0.1.0/c/graphK4Search.c [hunk 10]
```diff
@@ -9,6 +9,11 @@ See the LICENSE.TXT file for licensing information.
 #pragma clang diagnostic ignored "-Wstrict-prototypes"
 #pragma clang diagnostic ignored "-Wmissing-prototypes"
 #pragma clang diagnostic ignored "-Wdeprecated-non-prototype"
+#elif defined(__GNUC__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstrict-prototypes"
+#pragma GCC diagnostic ignored "-Wmissing-prototypes"
+
```
Comment (james-d-mitchell) [inline extern/edge-addition-planarity-suite-Version_3.0.1.0/c/graphK4Search.c:16]:
```suggestion
```

File: extern/edge-addition-planarity-suite-Version_3.0.1.0/c/graphK4Search_Extensions.c [hunk 11]
```diff
@@ -9,6 +9,11 @@ See the LICENSE.TXT file for licensing information.
 #pragma clang diagnostic ignored "-Wstrict-prototypes"
 #pragma clang diagnostic ignored "-Wmissing-prototypes"
 #pragma clang diagnostic ignored "-Wdeprecated-non-prototype"
+#elif defined(__GNUC__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstrict-prototypes"
+#pragma GCC diagnostic ignored "-Wmissing-prototypes"
+
```
Comment (james-d-mitchell) [inline extern/edge-addition-planarity-suite-Version_3.0.1.0/c/graphK4Search_Extensions.c:16]:
```suggestion
```

**Comment by james-d-mitchell** [comment]
Bunch of minor changes please @Joseph-Edwards, please remove the blank lines at the end of every `#if defined` pragma block (as partially indicated in the comments).

**Pull request merged**